compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/ProjectLoader.java

```diff
@@ -87,6 +91,43 @@ public static Project loadProject(Path path, ProjectEnvironmentBuilder projectEn
         return SingleFileProject.load(projectEnvironmentBuilder, absProjectPath);
     }
 
+    public static Optional<Path> findProjectRoot(Path filepath) throws ProjectException, IllegalArgumentException {
```

Comment:
This API should not be part of the ProjectLoader utility

Reply:
Can we move this into ProjectUtils ?

Reply:
How about creating a separate class called `ProjectPaths` and add all project path related utility APIs there? @azinneera @rasika

Reply:
AFAIU, the functionality of this method should be this: 
- First, check whether this path lies inside of a Ballerina project
- If true, return the root path of the project
- If false, return `Optional.empty()`;

if the above statement is correct, then the implementation should be simple IMO. Are there any other requirements here? @rasika @nadeeshaan @azinneera

Reply:
@sameerajayasoma yes, this should be the behaviour for both the single file and multi-module project. Also, I think it would be better if we can get this particular method out of the ProjectLoader.

Reply:
@sameerajayasoma +1 for the util. Can return root path of the project for both single-file and build projects, `Optional.empty()` otherwise.